libs/feature/editor/src/lib/components/record-form/form-field/form-field-constraints/form-field-constraints.component.ts

```diff
@@ -45,58 +43,25 @@ marker('editor.record.form.constraint.header.otherConstraints')
 })
 export class FormFieldConstraintsComponent implements OnInit {
   @Input() label: string
-  @Input() value: Constraint[] //constraintList
+  @Input() value: Constraint[]
   @Input() constraintType: CatalogRecordKeys
-  @Output() valueChange = new EventEmitter<Constraint[]>() //constraintListChange
-
-  legalConstraints$ = this.editorFacade.record$.pipe(
-    map((record) =>
-      'legalConstraints' in record ? record?.legalConstraints : []
-    )
-  )
-
-  securityConstraints$ = this.editorFacade.record$.pipe(
-    map((record) =>
-      'securityConstraints' in record ? record?.securityConstraints : []
-    )
-  )
-
-  otherConstraints$ = this.editorFacade.record$.pipe(
-    map((record) =>
-      'otherConstraints' in record ? record?.otherConstraints : []
-    )
-  )
+  @Output() valueChange = new EventEmitter<Constraint[]>()
 
   constraintsHeader = ''
   additionalConstraintsButtonLabel = ''
-  isAdditonalElementsVisible$: Observable<boolean>
 
   ngOnInit() {
     this.additionalConstraintsButtonLabel = `editor.record.form.constraint.add.${this.constraintType}`
     this.constraintsHeader = `editor.record.form.constraint.header.${this.constraintType}`
-
-    const constraintTypeObservableName = `${this.constraintType}$`
-    this.isAdditonalElementsVisible$ = this[constraintTypeObservableName].pipe(
-      map((constraints: Constraint[]) => constraints.length > 0)
-    )
   }
 
-  constructor(private editorFacade: EditorFacade) {}
-
-  handleURLChange(url: URL, index: number) {
-    const updatedConstraints = [...this.value]
-    updatedConstraints[index].url = url
-    console.log('updatedConstraints', updatedConstraints)
-    this.valueChange.emit(updatedConstraints)
-  }
-
-  handleConstraintTextChange(text: string, index: number) {
-    const updatedConstraints = [...this.value]
-    updatedConstraints[index].text = text
-    this.valueChange.emit(updatedConstraints)
+  handleConstraintChange(constraint: Constraint, index: number) {
+    this.value = [...this.value]
```

Comment:
Shouldn't we avoid assigning something to `this.value` as it is an `@Input`? Maybe instead we could do it in a similar way as in `handleConstraintsOrderChange`?

Reply:
good point, i'll try changing it and see if it works better :)